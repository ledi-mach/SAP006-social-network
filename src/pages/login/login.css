

.checkbox-keep-logged-in {
  font-size: 90% !important;
}


.login-container,
.register-container {
  height: 100vh;
  display: flex;
}

.login-container {
  background-image: url('../../images/login-bg.jpg');
  background-size: cover;
  background-repeat: no-repeat;
}

.register-container {
  background-image: url('../../images/register-bg.jpg');
  background-size: cover;
  background-repeat: no-repeat;
}

.login-left-section {
  position: relative;
  width: 35%;
  height: 95%;
  margin-top: 1.5%;
  margin-left: 33%;
  border-radius: 25px;
  background-color: rgba(255, 255, 255, 0.8);
}

.login-left-section header {
  margin-bottom: 2%;
  padding: 3% 4%;
  text-align: start;
  font-size: 90%;
  cursor: pointer;
}

.login-left-section main {
  position: absolute;
  top:15%;
}

.login-left-section img {
  width: 70%;
  margin-top: -5%;
  margin-bottom: 6%;
}

.login-left-section button,
.login-left-section input {
  font-family: 'Roboto', sans-serif;
  font-size: 110%;
  font-weight: 500;

  color: #025159;
  border-radius: 25px;
  box-shadow: 0 4px 20px rgba(10, 54, 37, 0.4);
  outline: none;
  cursor: pointer;
}

.login-btn-sig-log button {
  min-width: 30%;
  padding: 2% 2%;
  margin-bottom: 2%;
  margin-right: 2.5%;
  border: solid 0pt;
  color: white !important;
  background-color: var(--button-color);
}

.login-additionals button {
  font-size: 90%;
  border: 0pt;
  background-color: transparent;
  box-shadow: 0 0px 0px rgba(0, 0, 0, 0) !important;
}

.button-submit button {
  width: 15%;
  padding: 2% 0%;
  background-color: #025159;
  color: white;
  border-radius: 25px;
}


.login-left-section button:hover {
  transform: translateX(-2px);
}

.login-btn-sig-log button:hover {
  box-shadow: 0 4px 30px rgba(10, 54, 37, 0.95);
}

.login-left-section input {
  width: 60%;
  padding: 3% 3% 3% 3%;
  margin-bottom: 1%;
  border: solid 1pt;
}

.login-left-section form {
  position: relative;
}

.login-input-label {
  position: absolute;
  left: 22%;
  margin-top: 2%;
}

.login-left-section input:focus ~ .login-input-label,
.login-left-section input:valid ~ .login-input-label,
.login-input-label-up {
  font-size: 65%;
  margin-left: 1%;
  margin-top: 0.8%;
  color: grey;
}

.login-left-section input:focus {
  border: solid 2pt;
}
form input[type="checkbox"] {
  position: absolute;
  left: 40%;
  bottom: -2%;
  box-shadow: 0 0 0 0;
}

.transparent {
  color: transparent;
}

.login-input-has-an-error {
  border: #ff6666 solid 1.5pt !important;
  background-color: #ffe6e6;
}

.login-error-div {
  margin-top: 1%;
  margin-bottom: 1.5%;
}

.login-error {
  color: #ff3333;
  font-weight: 500;
}

.login-left-section footer {
  position: absolute;
  bottom:3%;
  left:41%;
  font-size: 80%;
}



@media screen and (max-width: 1024px) {
  .login-left-section{
    width: 60%;
    height: 50vh;
    margin: auto;
  }
  
  .login-left-section main{
    margin-left: 10%;
  }

  .login-left-section footer{
    left:43%
  } 

  form input[type="checkbox"]{
    margin-left:3%;
  }
}

@media screen and (max-width: 780px) {
  .login-left-section{
    height: 60vh;
  }
  
  .login-left-section main{
    margin-left: -2%;
  }

  .login-left-section footer{
    left:41%
  } 
}

@media screen and (max-width: 540px) {
  .login-left-section{
    height: 70vh;
  }

  .login-left-section footer{
    left:38%
  } 

  .login-btn-sig-log{
    margin-left: 5%;
  }

  form input[type="checkbox"]{
    margin-left:6%;
  }

}

@media screen and (max-width: 420px) {
  .login-left-section{
    width: 80%;
  }
  
  .login-left-section input {
    width: 75%;
    margin-top: 2%;
  }


.login-input-label {
  position: absolute;
  left: 15%;
  margin-top: 4%;
}

.login-left-section input:focus ~ .login-input-label,
.login-left-section input:valid ~ .login-input-label,
.login-input-label-up {
  font-size: 65%;
  margin-left: 2%;
  margin-top: 3%;
}

}

@media screen and (max-width: 360px) {
  .login-left-section{
    height: 80vh;
  }

  .login-left-section footer{
    left:35%
  } 

  form input[type="checkbox"]{
    margin-left:13%;
    top:80%;
  }
}

@media screen and (max-width: 280px) {
  .login-left-section{
    height: 70vh;
  }

  form input[type="checkbox"]{
    margin-left:18%;
  }

}

@media screen and (max-height: 450px) {
  footer {
  display: none;
  }
  
}