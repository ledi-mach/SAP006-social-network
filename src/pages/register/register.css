
.register-container {
  background-size: cover;
  background-repeat: no-repeat;
  height: 32vh;
}

@media screen and (max-width: 1024px) {
  .register-container{
    height: 100vh;
  }
}

@media screen and (max-width: 375px) {
  .register-container input[type="checkbox"]{
    margin-left: 10%;
    top:86%;
  }

  .login-btn-sig-log button{
    padding:3%
  }
}

@media screen and (max-width: 320px) {
  .register-container input[type="checkbox"]{
    margin-left: 13%;
  }
}

@media screen and (max-width: 280px) {
  .register-container input[type="checkbox"]{
    margin-left: 17%;
  }
}