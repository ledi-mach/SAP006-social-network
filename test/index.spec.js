/**
* @jest-environment jsdom
*/
import * as auth from '../src/lib/auth.js';
import { Login } from '../src/pages/login/login.js';
import { Register } from '../src/pages/register/register.js';
import {
  loginWithGoogle,
  loginWithEmailAndPassword,
  registerAccount,
  resetPassword,
  getTheRoad,
}
  from '../src/lib/auth.js';

jest.mock('../src/lib/auth.js');

describe('Render página de login', () => {
  it('carrega a página', () => {
    expect(Login()).toMatchSnapshot();
  });
  it('click no botão de login com o Google', () => {
    loginWithGoogle.mockResolvedValueOnce(true);
    const pageLogin = Login();
    pageLogin.querySelector('#btn-login-with-google').dispatchEvent(new Event('click'));
    expect(auth.loginWithGoogle).toBeCalled();
  });
});

describe('loginEmailAndPassword', () => {
  it('deve logar com email e senha', () => {
    loginWithEmailAndPassword.mockResolvedValueOnce(true);
    const pageLogin = Login();
    pageLogin.querySelector('#input-email').value = 'teste@teste.com';
    pageLogin.querySelector('#input-password').value = '12345678';
    const checkboxKeepLoggedIn = pageLogin.querySelector('#checkbox-keep-logged-in');
    pageLogin.querySelector('#btn-login').dispatchEvent(new Event('click'));
    expect(auth.loginWithEmailAndPassword).toBeCalled();
    expect(auth.loginWithEmailAndPassword).toHaveBeenCalledWith('teste@teste.com', '12345678', checkboxKeepLoggedIn);
  });
});

describe('resetPassword', () => {
  it('Deverá enviar um e-mail ao usuário para a restauração de senha.', () => {
    resetPassword.mockResolvedValueOnce(true);
    const pageLogin = Login();
    pageLogin.querySelector('#input-email').value = 'teste@teste.com';
    pageLogin.querySelector('#btn-forgot-password').dispatchEvent(new Event('click'));
    expect(auth.resetPassword).toBeCalled();
    expect(auth.resetPassword).toHaveBeenCalledWith('teste@teste.com');
  });
});
describe('getTheRoad', () => {
  it('Deverá redirecionar ao Feed.', () => {
    getTheRoad.mockResolvedValueOnce(true);
    const pageLogin = Login();
    pageLogin.querySelector('#btn-sign-in').dispatchEvent(new Event('click'));
    expect(auth.getTheRoad).toBeCalled();
  });
});

describe('registerAccount', () => {
  it('Deverá criar uma conta.', () => {
    registerAccount.mockResolvedValueOnce(true);
    const pageRegister = Register();
    pageRegister.querySelector('#input-email').value = 'teste@teste.com';
    pageRegister.querySelector('#input-password').value = '12345678';
    pageRegister.querySelector('#input-name').value = 'Teste';
    const checkboxKeepLoggedIn = pageRegister.querySelector('#checkbox-keep-logged-in');
    pageRegister.querySelector('#btn-register-account').dispatchEvent(new Event('click'));
    expect(auth.registerAccount).toBeCalled();
    expect(auth.registerAccount).toHaveBeenCalledWith('teste@teste.com', '12345678', 'Teste', checkboxKeepLoggedIn);
  });
});
