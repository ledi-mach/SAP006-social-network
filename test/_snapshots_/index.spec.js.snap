// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section class="login-left-section">
    <header>
      <a>Sobre n√≥s</a>
    </header>
    <main>
      <div>
        <img src="./images/logo-login.png"/>
      </div>
      <form>
        <div class="div-email" id="div-email">
          <input required type="email" class="input-email" id="input-email">
          <label class="login-input-label" id="label-input-email">Email</label>
        </div>
        <div class="div-password" id="div-password">
          <input required type="password" class="input-password" id="input-password">
            <img class="eye" id="eye" src="./images/eye-off.png"/>
            <label class="login-input-label">Senha</label> 
          <br>
          <label class="checkbox-keep-logged-in"> Mantenha-me logadex
            <input type="checkbox" id="checkbox-keep-logged-in">
          </label>
        </div>
      </form>
      <div class="login-error-div" id="print-error-here"> 
        <p class="transparent"> . </p>
      </div>
      <div class="login-btn-sig-log">
        <button id="btn-login" class="btn-login">ENTRAR</button>  
        <button id="btn-sign-in">CADASTRAR</button>
      </div>
      <div class="login-additionals">
        <button id="btn-login-with-google"> Ou conecte-se com o gmail</button>
        <br>
        <button id="btn-forgot-password"> Esqueceu a senha?</button>
      </div>
    </main>
    <footer> Willing &copy 2021</footer>
  </section> 
`;
